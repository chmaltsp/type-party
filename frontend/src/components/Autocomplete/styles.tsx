import styled from 'sc';

export const ListWrapper = styled.ul`
  margin-top: 0;
  padding-left: 0;
`;

export interface ListItemProps {
  isActive?: boolean;
  isSelected?: boolean;
  value: string;
}
export const ListItem = styled.li<ListItemProps>`
  border: 1px solid ${({ theme }) => theme.colors.black};
  border-top: 0;
  list-style: none;
  text-decoration: none;
  color: ${({ isSelected, theme }) =>
    isSelected ? theme.colors.greyC4C : theme.colors.black};
  padding: ${({ theme }) => theme.spacing.xs}px;
  background-color: ${({ isActive, theme }) =>
    isActive ? theme.colors.greyC4C : 'transparent'};
  &:last-child {
    border-radius: ${({ theme }) => theme.borderRadius};
  }
`;

export const TagSpacer = styled.div`
  display: grid;
  grid-gap: 6;
  margin: 0 4px 4px;
  grid-auto-flow: column;
  align-items: center;
`;
