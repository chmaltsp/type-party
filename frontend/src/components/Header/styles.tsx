import { em, transparentize } from 'polished';

import styled, { media } from 'sc';

import { Star as StarBase } from '../Images';

import LinkBase from '../Link';
import LogoBase from '../Logo';

const Link = styled(LinkBase)`
  font-size: ${em(18)};
  font-weight: 400;
  text-decoration: none;
  margin-right: ${({ theme }) => em(theme.spacing.md)};
`;

const Wrapper = styled.div`
  display: flex;
  margin: 0 ${em(136)};
  padding-top: ${em(48)};
  padding-bottom: ${em(16)};
  border-bottom: none;
  ${media.tablet`
  border-bottom: 1px solid ${(props: any) =>
    transparentize(0.8, props.theme.colors.black)};
`};
  ${media.desktop`
  margin: 0 ${em(32)};
`};
`;

const Star = styled(StarBase)`
  display: none;
  width: 48px;
  height: 48px;
  background-repeat: no-repeat;
  ${media.tablet`
    display: inline-block;
  `};
`;

const Logo = styled(LogoBase)`
  display: block;
  width: ${em(80)};
  ${media.tablet`
    display: none;
  `}
  ${media.desktop`
    width: ${em(72)};
    height: ${em(72)};
  `};
`;

const NavlinksWrapper = styled.div`
  flex: 1;
  text-align: right;
  ${media.tablet`
      display: none;
  `};
`;

const MenuLink = styled.div`
  font-size: ${em(18)};
  font-weight: 400;
  margin-left: auto;
  display: none;
  align-self: center;
  cursor: pointer;
  ${media.tablet`
      display: inline-block;
  `};
`;

export { MenuLink, NavlinksWrapper, Logo, Star, Wrapper, Link };
