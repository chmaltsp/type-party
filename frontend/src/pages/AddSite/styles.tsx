import { Form as FormBase } from 'formik';
import styled from 'sc';
import ButtonBase, { LoadingButton } from '../../components/Button';
import Flex from '../../components/Flex';

export const LeftColumn = styled(Flex)`
  flex: 1;
  max-width: 469px;
  flex-direction: column;
`;
export const AddTypefaceButton = styled(ButtonBase)`
  /* width: 200px; */
`;
export const RightColumn = styled(Flex)`
  margin-left: ${({ theme }) => theme.baseSpacing * 13 + 'px'};
  max-width: 569px;
  flex-direction: column;
  flex: 1;
`;
export const Form = styled(FormBase)`
  margin: ${props => props.theme.baseSpacing * 2}px auto;
  display: flex;
  flex-direction: row;
  width: 100%;
`;
export const SubmitButton = styled(LoadingButton)`
  /* background-color: transparent; */
  border: 2px ${({ theme }) => theme.colors.black} solid;
  margin: ${({ theme }) => theme.baseSpacing * 2}px 0;
  border-radius: ${({ theme }) => theme.baseSpacing / 2}px;
  height: 64px;
  /* color: ${({ theme }) => theme.colors.black}; */
  cursor: pointer;
  /* padding: ${({ theme }) => theme.baseSpacing}px ${({ theme }) =>
  theme.baseSpacing * 2}px; */
`;
