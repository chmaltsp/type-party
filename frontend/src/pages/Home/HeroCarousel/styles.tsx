import { em } from 'polished';
import {
  ButtonBack as ButtonBackBase,
  ButtonNext as ButtonNextBase,
  DotGroup as DotGroupBase,
} from 'pure-react-carousel';
import { css, media } from 'sc';
import styled from 'styled-components';
import CardBase from '../../../components/Card';
export const Wrapper = styled.div`
  width: 100%;
  margin: 0 auto;
  position: relative;
`;
export const LeftArrow = styled.img``;
export const RightArrow = styled.img``;
const buttonNormal = css`
  top: 40%;
  position: absolute;
  border: 0;
  background: 0;
  &:focus {
    outline: none;
  }
  ${media.tablet`
     display: none;
  `}
`;
export const ButtonBack = styled(ButtonBackBase)`
  ${buttonNormal}
  left: 50px;
  @media screen and (max-width: 1248px) {
    left: 25px;
  }
`;
export const ButtonNext = styled(ButtonNextBase)`
  ${buttonNormal}
  right: 50px;
  @media screen and (max-width: 1248px) {
    right: 25px;
  }
`;
export const DotGroup = styled(DotGroupBase)`
  margin: 0 auto;
  text-align: center;
  padding-top: 32px;
  > button {
    margin-right: 8px;
    border: 1px #000 solid;
    border-radius: 50%;
    background: 0;
    width: 16px;
    height: 16px;
    &:focus {
      outline: none;
    }
    &.carousel__dot--selected {
      background: #000;
    }
  }
`;
export const Card = styled(CardBase)`
  ${media.desktop`
      max-width: ${em(600)};
  `}
 
  @media screen and (max-width: 1248px) {
    max-width: ${em(850)};
    padding: 0 ${em(100)};
  }
  ${media.tablet`
      max-width: ${em(500)};
      padding: 0 ${em(32)};
  `}
  max-width: ${em(976)};
  margin: 0 auto;
`;
