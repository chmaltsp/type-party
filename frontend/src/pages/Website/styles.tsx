import styled, { media } from 'sc';
import { ButtonA } from '../../components/Button';
import { Container as ContainerBase } from '../../components/Container';
import Flex from '../../components/Flex';
import PageTitle from '../../components/PageTitle';
export const DetailWrapper = styled(Flex)`
  flex-direction: row;
  ${media.tablet`
    flex-direction: column;
    margin: ${({ theme }: any) => theme.spacing.md}px 0;
  `}
`;
export const Image = styled.img`
  flex: 1;
  width: 50%;
  height: 419px;
  background-color: ${({ theme }) => theme.colors.greyC4C};
  ${media.tablet`
    width: 100%;
  `};
`;

export const WebsiteInfo = styled(Flex)`
  flex: 1;
  padding: 0 ${({ theme }) => theme.spacing.md}px;
  width: 50%;
  flex-direction: column;
  ${media.tablet`
    width: 100%;
  `};
`;
export const Title = styled(PageTitle)`
  margin-bottom: ${({ theme }) => theme.spacing.md}px;
`;
export const Button = styled(ButtonA)`
  margin-top: ${({ theme }) => theme.spacing.lg}px;
  display: inline-block;
`;
export const Line = styled.hr`
  width: 100%;
  color: ${({ theme }) => theme.colors.black};
  opacity: 0.2;
  margin-top: ${props => props.theme.baseSpacing * 12}px;
  margin-bottom: 0;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
`;
export const Container = styled(ContainerBase)`
  margin-bottom: ${props => props.theme.baseSpacing * 15}px;
`;
export const TagWrapper = styled.div`
  margin-top: ${({ theme }) => theme.spacing.md}px;
  display: flex;
  flex-wrap: wrap;
  > div {
    margin-right: ${({ theme }) => theme.spacing.sm}px;
    margin-top: ${({ theme }) => theme.spacing.sm}px;
  }
`;
