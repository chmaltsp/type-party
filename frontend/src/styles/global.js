import { normalize } from 'polished';

import { injectGlobal } from 'sc';

// @ts-ignore
export default injectGlobal`
    @import url('https://fonts.googleapis.com/css?family=Muli');

    ${normalize()};

    body {
        font-family: 'Muli', sans-serif;
        height: 100%;
        font-size: 16px;
        width: 100%;
        background-color: #FFF;
    }
`;
