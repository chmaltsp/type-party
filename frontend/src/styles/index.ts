import * as styledComponents from 'styled-components';

// tslint:disable-next-line
import { ThemedStyledComponentsModule } from 'styled-components';

import theme from './theme';

const {
  default: styled,
  css,
  injectGlobal,
  keyframes,
  ThemeProvider,
} = styledComponents as ThemedStyledComponentsModule<typeof theme>;

export { css, injectGlobal, keyframes, ThemeProvider };
export default styled;
