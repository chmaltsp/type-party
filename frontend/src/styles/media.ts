// these sizes are arbitrary and you can set them to whatever you wish
import { css } from './';

const sizes = {
  giant: 1170,
  // tslint:disable-next-line:object-literal-sort-keys
  desktop: 992,
  tablet: 768,
  phone: 376,
};

interface Media {
  giant?: any;
  desktop?: any;
  tablet?: any;
  phone?: any;
}

// iterate through the sizes and create a media template
export const media: Media = Object.keys(sizes).reduce(
  (accumulator: Media, label: string) => {
    // use em in breakpoints to work properly cross-browser and support users
    // changing their browsers font-size: https://zellwk.com/blog/media-query-units/
    const emSize = sizes[label] / 16;
    accumulator[label] = (templateStrings: any, ...args: any[]) => css`
    @media (max-width: ${emSize}em) {
      ${css(templateStrings, ...args)}
    }
  `;
    return accumulator;
  },
  {}
);
