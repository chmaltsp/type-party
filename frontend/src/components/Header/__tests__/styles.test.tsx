import * as React from 'react';

import { Link, Logo, MenuLink, NavlinksWrapper, Star, Wrapper } from '../styles';

import { shallowWithTheme } from 'sc';

describe('<Link', () => {
  it('should render properly', () => {
    expect(shallowWithTheme(<Link to="/non-route" />)).toMatchSnapshot();
  });
});

describe('<Logo', () => {
  it('should render properly', () => {
    expect(shallowWithTheme(<Logo />)).toMatchSnapshot();
  });
});

describe('<MenuLink', () => {
  it('should render properly', () => {
    expect(shallowWithTheme(<MenuLink />)).toMatchSnapshot();
  });
});

describe('<NavlinksWrapper', () => {
  it('should render properly', () => {
    expect(shallowWithTheme(<NavlinksWrapper />)).toMatchSnapshot();
  });
});

describe('<Star', () => {
  it('should render properly', () => {
    expect(shallowWithTheme(<Star />)).toMatchSnapshot();
  });
});

describe('<Wrapper', () => {
  it('should render properly', () => {
    expect(shallowWithTheme(<Wrapper />)).toMatchSnapshot();
  });
});
