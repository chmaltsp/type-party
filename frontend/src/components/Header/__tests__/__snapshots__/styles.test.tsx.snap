// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link should render properly 1`] = `
.c0 {
  font-size: 1.125em;
  font-weight: 400;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-right: 1.5em;
}

<Styled(Link)
  className="c0"
  to="/non-route"
/>
`;

exports[`<Logo should render properly 1`] = `
.c0 {
  max-width: 6.25em;
  width: 100%;
  height: auto;
  display: block;
  width: 5em;
}

@media (max-width:48em) {
  .c0 {
    display: none;
  }
}

@media (max-width:62em) {
  .c0 {
    width: 4.5em;
    height: 4.5em;
  }
}

<img
  className="c0"
  src="test-file-stub"
/>
`;

exports[`<MenuLink should render properly 1`] = `
.c0 {
  font-size: 1.125em;
  font-weight: 400;
  margin-left: auto;
  display: none;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  cursor: pointer;
}

@media (max-width:48em) {
  .c0 {
    display: inline-block;
  }
}

<div
  className="c0"
/>
`;

exports[`<NavlinksWrapper should render properly 1`] = `
.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-align: right;
}

@media (max-width:48em) {
  .c0 {
    display: none;
  }
}

<div
  className="c0"
/>
`;

exports[`<Star should render properly 1`] = `
.c0 {
  width: 300px;
  background-image: url(test-file-stub);
  display: none;
  width: 48px;
  height: 48px;
  background-repeat: no-repeat;
}

@media (max-width:48em) {
  .c0 {
    display: inline-block;
  }
}

<div
  className="c0"
/>
`;

exports[`<Wrapper should render properly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0 8.5em;
  padding-top: 3em;
  padding-bottom: 1em;
  border-bottom: none;
}

@media (max-width:48em) {
  .c0 {
    border-bottom: 1px solid rgba(0,0,0,0.2);
  }
}

@media (max-width:62em) {
  .c0 {
    margin: 0 2em;
  }
}

<div
  className="c0"
/>
`;
